scripts/doctest.awk

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env -S gawk -f
+#
+# doctest.awk formats all evy code blocks in a markdown file and executes that
+# code feeding it the specified input and replacing the output code block with
+# the output of running that evy code.
+#
+# Evy code is identified by a triple-backtick code block with the language
+# specified as "evy". Input that should be provided to the evy program on
+# stdin should follow in another code block with the language specified as
+# "evuy:input". If a code block with the language "evy:output" follows,
+# its contents will be replaced with the output of executing the evy program.
+# If the output contains the escape sequence for clearing the terminal, that
+# and all preceding output are discarded.
+#
+# e.g.
+#
+# ```evy
+# read name
+# print "Hello" name
+# ```
+#
+# ```evy:input
+# Fox
+# ```
+#
+# ```evy:output
+# Hello Fox
+# ```
+#
+# This script requires gawk. It can be run with "-i inplace" in which
```

Comment:
is there something that can be done about the gawk dependency? it is the only global dependency that this repo has apart from git i think.

Reply:
I use a gawk feature - coprocessors - using the `|&` pipe form that allows me to send input to a program and read it back. I can probably do it with temporary files instead so it should work with the standard Mac awk (although the man page on macos is terrible and underdocumented).

Reply:
Oh, I also use gawk for the `-i inplace` CLI syntax to replace the contents of a file rather than writing to stdout. I guess that's not needed too - the when the makefile runs this, it just has to do the tempfile dance.

Reply:
Ok, I've removed the gawk-specific code and removed the comment saying to use `-i inplace`. This should work with regular/posix awk and hopefully works on a mac. You need to run `scripts/doctest.awk docs/builtins.md > new.md` and then `mv new.md docs/builtins.mk` - the makefile will need to do this (perhaps with a tmp name rather that `new.md`)